tough/src/editor/mod.rs

```diff
@@ -125,15 +146,30 @@ impl RepositoryEditor {
     /// While `RepositoryEditor`s fields are all `Option`s, this step requires,
     /// at the very least, that the "version" and "expiration" field is set for
     /// each role; e.g. `targets_version`, `targets_expires`, etc.
-    pub fn sign(self, keys: &[Box<dyn KeySource>]) -> Result<SignedRepository> {
+    pub fn sign(mut self, keys: &[Box<dyn KeySource>]) -> Result<SignedRepository> {
         let rng = SystemRandom::new();
         let root = KeyHolder::Root(self.signed_root.signed.signed.clone());
+        // Sign the targets editor if able to with the provided keys
+        self.sign_targets_editor(keys)?;
+        let targets = self.signed_targets.clone().context(error::NoTargets)?;
+        let delegated_targets = targets.signed.signed_delegated_targets();
+        let signed_targets = SignedRole::from_signed(targets)?;
+
+        let signed_delegated_targets = if delegated_targets.is_empty() {
+            None
+        } else {
+            let mut roles = Vec::new();
+            for role in delegated_targets {
+                roles.push(SignedRole::from_signed(role)?)
+            }
+            Some(SignedDelegatedTargets {
+                roles,
+                consistent_snapshot: self.signed_root.signed.signed.consistent_snapshot,
```

Comment:
Let's add some comments here describing why the delegated targets process is slightly different.